integrationtests/mayavi/common.py

```diff
@@ -383,12 +386,18 @@ def run_standalone(self):
             type, value, tb = self.exception_info
             raise type, value, tb
 
+
     def run(self):
         """This starts everything up and runs the test.  Call main to
         run the test."""
         # Calls the users test code.
         try:
-            self.do()
+            if self.profile:
+                memory_assistant = MemoryAssistant()
+                memory_assistant.assertReturnsMemory(self.do_profile,
```

Comment:
The reason to have a separate method `do_profile` is that the the method `do` also does a series of checks/assertions. We do not want to do that while checking for memory usage/leaks. In the earlier version of `test_memory_usage`, we found that reusing `do` for memory usage etc resulted in less readable code due to the assertions, saving/loading vis etc.